File: app/src/main/java/in/gov/mahapocra/mahavistaarai/ui/adapters/ClimateResilientTechnologyAdapter.java
package in.gov.mahapocra.mahavistaarai.ui.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import in.co.appinventor.services_api.listener.OnMultiRecyclerItemClickListener;
import in.gov.mahapocra.mahavistaarai.R;

public class ClimateResilientTechnologyAdapter extends RecyclerView.Adapter<ClimateResilientTechnologyAdapter.ViewHolder> {

    private final OnMultiRecyclerItemClickListener listener;
    private final Context mContext;
    private final JSONArray mJSONArray;

    public ClimateResilientTechnologyAdapter(Context context, OnMultiRecyclerItemClickListener listener, JSONArray jsonArray) {
        this.mContext = context;
        this.mJSONArray = jsonArray;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ClimateResilientTechnologyAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.single_climate_resilient_tech, parent, false);
        return new ClimateResilientTechnologyAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ClimateResilientTechnologyAdapter.ViewHolder holder, int position) {
        try {
            holder.setDvata(mJSONArray.getJSONObject(position), listener);
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    @Override
    public int getItemCount() {
        if (mJSONArray != null) {
            return mJSONArray.length();
        } else {
            return 0;
        }
    }
    public class ViewHolder extends RecyclerView.ViewHolder {
        private final TextView nameTextView2;
        private final TextView nameTextView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            nameTextView2 = itemView.findViewById(R.id.nameTextView2);
            nameTextView = itemView.findViewById(R.id.nameTextView);
        }
        public void setDvata(JSONObject jsonObject, OnMultiRecyclerItemClickListener listener) {
            try {
                nameTextView2.setText(jsonObject.getString("MainGroupSRNo")+" - " );
                nameTextView.setText( jsonObject.getString("MainGroupName"));
            } catch (JSONException e) {
                e.printStackTrace();
            }

            itemView.setOnClickListener(v -> listener.onMultiRecyclerViewItemClick(2, jsonObject));
        }
    }
}

File: app/src/main/java/in/gov/mahapocra/mahavistaarai/ui/adapters/ClimateResilientTechnologyAdapter.kt
package `in`.gov.mahapocra.mahavistaarai.ui.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import `in`.co.appinventor.services_api.listener.OnMultiRecyclerItemClickListener
import `in`.gov.mahapocra.mahavistaarai.R
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

class ClimateResilientTechnologyAdapter(
    private val mContext: Context?,
    private val listener: OnMultiRecyclerItemClickListener,
    private val mJSONArray: JSONArray?
) : RecyclerView.Adapter<ClimateResilientTechnologyAdapter.ViewHolder?>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(mContext).inflate(R.layout.sop_category_item_view, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        try {
            holder.setData(mJSONArray!!.getJSONObject(position), listener)
        } catch (e: JSONException) {
            e.printStackTrace()
        }
    }

    override fun getItemCount(): Int {
        return mJSONArray?.length() ?: 0
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val nameTextView2: TextView = itemView.findViewById(R.id.nameTextView2)
        private val nameTextView: TextView = itemView.findViewById(R.id.nameTextView)

        fun setData(jsonObject: JSONObject, listener: OnMultiRecyclerItemClickListener) {
            try {
                nameTextView2.text = jsonObject.getString("MainGroupSRNo")
                nameTextView.text = jsonObject.getString("MainGroupName")
            } catch (e: JSONException) {
                e.printStackTrace()
            }
            itemView.setOnClickListener { listener.onMultiRecyclerViewItemClick(2, jsonObject) }
        }
    }
}
